elements/lisk-p2p/src/utils/validate.ts

```diff
@@ -123,74 +117,53 @@ export const validatePeerAddress = (ip: string, wsPort: number): boolean => {
 	return true;
 };
 
-export const validatePeerInfoSchema = (rawPeerInfo: unknown): P2PPeerInfo => {
+export const validatePeerInfo = (
+	rawPeerInfo: unknown,
+	maxByteSize: number,
+): P2PPeerInfo => {
 	if (!rawPeerInfo) {
-		throw new InvalidPeerError(`Invalid peer object`);
+		throw new InvalidPeerInfoError(`Invalid peer object`);
 	}
 
 	const protocolPeer = rawPeerInfo as ProtocolPeerInfo;
-	const ipAddress = protocolPeer.ip || protocolPeer.ipAddress;
 
 	if (
-		!ipAddress ||
+		!protocolPeer.ipAddress ||
 		!protocolPeer.wsPort ||
-		!validatePeerAddress(ipAddress, protocolPeer.wsPort)
+		!validatePeerAddress(protocolPeer.ipAddress, protocolPeer.wsPort)
 	) {
-		throw new InvalidPeerError(
-			`Invalid peer ip or port for peer with ip: ${ipAddress} and wsPort ${
-				protocolPeer.wsPort
-			}`,
+		throw new InvalidPeerInfoError(
+			`Invalid peer ipAddress or port for peer with ip: ${
+				protocolPeer.ipAddress
+			} and wsPort ${protocolPeer.wsPort}`,
 		);
 	}
 
-	if (!protocolPeer.version || !isValidVersion(protocolPeer.version)) {
-		throw new InvalidPeerError(
-			`Invalid peer version for peer with ip: ${protocolPeer.ip}, wsPort ${
-				protocolPeer.wsPort
-			} and version ${protocolPeer.version}`,
+	const peerInfo = sanitizeIncomingPeerInfo(protocolPeer);
```

Comment:
If this is incoming, the validation must be too. Is it possible to extract this making the validation in a different step than sanitization?